epictrack-web/src/components/project/ProjectForm.tsx

```diff
@@ -244,17 +256,22 @@ export default function ProjectForm({ ...props }) {
                 }}
               >
                 <ETFormLabel required>Proponent</ETFormLabel>
-                <ETFormLabel>
-                  <Box
-                    sx={{
-                      opacity: disabled ? "100" : "0",
-                      cursor: "pointer",
-                    }}
-                    component="img"
-                    src={LockClosed}
-                    alt="Lock"
-                  />
-                </ETFormLabel>
+                <When condition={disabled}>
+                  <If condition={specialField === "proponent_id"}>
+                    <Then>
+                      <IconButton onClick={() => setSpecialField("")}>
```

Comment:
if the difference is only in the onClick can we just add a ternary there instead of wrapping the whole button component